cmake/modules/FindAbseil.cmake

```diff
@@ -0,0 +1,79 @@
+  set(ABSEIL_INCLUDE_DIRS ${ABSEIL_PROJECT_SRC_DIR})
+  set(ABSEIL_LIBRARY_PATH
+    "${ABSEIL_PROJECT_BUILD_DIR}/absl/synchronization/${CMAKE_STATIC_LIBRARY_PREFIX}absl_synchronization${CMAKE_STATIC_LIBRARY_SUFFIX}")
+  set(dependent_libraries
+    "${ABSEIL_PROJECT_BUILD_DIR}/absl/time/libabsl_time.a"
+    "${ABSEIL_PROJECT_BUILD_DIR}/absl/algorithm/libabsl_algorithm.a"
+    "${ABSEIL_PROJECT_BUILD_DIR}/absl/base/libabsl_base.a"
+    "${ABSEIL_PROJECT_BUILD_DIR}/absl/base/libabsl_dynamic_annotations.a"
+    "${ABSEIL_PROJECT_BUILD_DIR}/absl/base/libabsl_malloc_internal.a"
+    "${ABSEIL_PROJECT_BUILD_DIR}/absl/base/libabsl_spinlock_wait.a"
+    "${ABSEIL_PROJECT_BUILD_DIR}/absl/base/libabsl_throw_delegate.a"
+    "${ABSEIL_PROJECT_BUILD_DIR}/absl/container/libabsl_container.a"
+    "${ABSEIL_PROJECT_BUILD_DIR}/absl/container/libtest_instance_tracker_lib.a"
+    "${ABSEIL_PROJECT_BUILD_DIR}/absl/debugging/libabsl_debugging.a"
+    "${ABSEIL_PROJECT_BUILD_DIR}/absl/debugging/libabsl_examine_stack.a"
+    "${ABSEIL_PROJECT_BUILD_DIR}/absl/debugging/libabsl_failure_signal_handler.a"
+    "${ABSEIL_PROJECT_BUILD_DIR}/absl/debugging/libabsl_leak_check.a"
+    "${ABSEIL_PROJECT_BUILD_DIR}/absl/debugging/libabsl_stack_consumption.a"
+    "${ABSEIL_PROJECT_BUILD_DIR}/absl/debugging/libabsl_stacktrace.a"
+    "${ABSEIL_PROJECT_BUILD_DIR}/absl/debugging/libabsl_symbolize.a"
+    "${ABSEIL_PROJECT_BUILD_DIR}/absl/memory/libabsl_memory.a"
+    "${ABSEIL_PROJECT_BUILD_DIR}/absl/meta/libabsl_meta.a"
+    "${ABSEIL_PROJECT_BUILD_DIR}/absl/numeric/libabsl_int128.a"
+    "${ABSEIL_PROJECT_BUILD_DIR}/absl/numeric/libabsl_numeric.a"
+    "${ABSEIL_PROJECT_BUILD_DIR}/absl/strings/libabsl_strings.a"
+    "${ABSEIL_PROJECT_BUILD_DIR}/absl/types/libabsl_bad_any_cast.a"
+    "${ABSEIL_PROJECT_BUILD_DIR}/absl/types/libabsl_any.a"
+    "${ABSEIL_PROJECT_BUILD_DIR}/absl/types/libabsl_span.a"
+    "${ABSEIL_PROJECT_BUILD_DIR}/absl/types/libabsl_variant.a"
+    "${ABSEIL_PROJECT_BUILD_DIR}/absl/types/libabsl_optional.a"
+    "${ABSEIL_PROJECT_BUILD_DIR}/absl/types/libabsl_bad_optional_access.a"
+    "${ABSEIL_PROJECT_BUILD_DIR}/absl/utility/libabsl_utility.a"
+    )
+  ExternalProject_Add(${ABSEIL_PROJECT_NAME}
+    PREFIX ${ABSEIL_PROJECT_NAME}
+    GIT_REPOSITORY   https://github.com/abseil/abseil-cpp.git
+    GIT_TAG          44aa275286baf97fc13529aca547a88b180beb08
+    INSTALL_COMMAND  ""
+    CMAKE_CACHE_ARGS "-DCMAKE_POSITION_INDEPENDENT_CODE:BOOL=ON;-DBUILD_TESTING:BOOL=OFF;-DCMAKE_BUILD_TYPE:STRING=Release"
+    BUILD_BYPRODUCTS "${ABSEIL_LIBRARY_PATH};${dependent_libraries}"
+  )
+  add_library(standalone_absl STATIC IMPORTED)
+  set_target_properties(standalone_absl
+    PROPERTIES INTERFACE_INCLUDE_DIRECTORIES
+    ${ABSEIL_INCLUDE_DIRS}
+  )
+  set_target_properties(standalone_absl
+    PROPERTIES IMPORTED_LOCATION
+    ${ABSEIL_LIBRARY_PATH}
+    INTERFACE_LINK_LIBRARIES
+    "${dependent_libraries}"
```

Comment:
@hauke76 Is there a cleaner way to glob all static libraries into one target?
ABSEIL_LIBRARY_PATH=synchronization/libabsl_synchronization.a was an arbitrary choice only because this is the first one we will use.

Reply:
As discussed offline, this is fine.